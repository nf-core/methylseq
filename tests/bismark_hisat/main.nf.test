import com.askimed.nf.test.util.FileUtil;

nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "Bismark Hisat"

    test("Bismark Hisat Single End") {
        when {
            params {
                aligner = "bismark_hisat"
                save_reference = true
                outdir = "$outputDir"
            }
        }

        then {
            // Clean up anything from previous runs
            FileUtil.deleteDirectory(new File(".nf-test/indexes/"));
            new File(".nf-test/indexes/").mkdirs()
            // Copy the ref genome to a stable location for next tests
            FileUtil.copyDirectory(
                new File("$outputDir/bismark_hisat/reference_genome/BismarkIndex/").getAbsolutePath(),
                new File(".nf-test/indexes/BismarkIndex/").getAbsolutePath()
            )

            assertAll(
                { assert workflow.success },
                { assert snapshot(path("$outputDir/bismark_hisat/methylation_calls/splitting_report/Ecoli_10K_methylated_1_val_1_bismark_hisat2_pe.deduplicated_splitting_report.txt"),
                            path("$outputDir/bismark_hisat/methylation_calls/splitting_report/SRR389222_sub1_trimmed_bismark_hisat2.deduplicated_splitting_report.txt"),
                            path("$outputDir/bismark_hisat/methylation_calls/splitting_report/SRR389222_sub2_trimmed_bismark_hisat2.deduplicated_splitting_report.txt"),
                            path("$outputDir/qualimap/Ecoli_10K_methylated/genome_results.txt"),
                            path("$outputDir/qualimap/SRR389222_sub1/genome_results.txt"),
                            path("$outputDir/qualimap/SRR389222_sub2/genome_results.txt"),
                            path("$outputDir/pipeline_info/software_versions.yml")).match("single_end") },
                { assert new File("$outputDir/bismark_hisat/summary/bismark_summary_report.txt").exists() }            
            )
        }
    }

    test("Bismark Hisat Single End With Index") {
        when {
            params {
                aligner = "bismark_hisat"
                // Generated by previous test
                bismark_index = ".nf-test/indexes/BismarkIndex/"
                local_alignment = true
                outdir = "$outputDir"
            }
        }

        then {
            assert workflow.success
        }
    }

    test("Bismark Hisat Single End With RRBS") {
        when {
            params {
                aligner = "bismark_hisat"
                save_reference = true
                skip_trimming = true
                rrbs = true
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(path("$outputDir/bismark_hisat/methylation_calls/splitting_report/Ecoli_10K_methylated_R1_bismark_hisat2_pe_splitting_report.txt"),
                            path("$outputDir/bismark_hisat/methylation_calls/splitting_report/SRR389222_sub1_bismark_hisat2_splitting_report.txt"),
                            path("$outputDir/bismark_hisat/methylation_calls/splitting_report/SRR389222_sub2.merged_bismark_hisat2_splitting_report.txt"),
                            path("$outputDir/qualimap/Ecoli_10K_methylated/genome_results.txt"),
                            path("$outputDir/qualimap/SRR389222_sub1/genome_results.txt"),
                            path("$outputDir/qualimap/SRR389222_sub2/genome_results.txt"),
                            path("$outputDir/pipeline_info/software_versions.yml")).match("single_end_rrbs") },
                { assert new File("$outputDir/bismark_hisat/summary/bismark_summary_report.txt").exists() }            
            )
        }
    }

    test("Bismark Hisat Single End With RRBS With Index") {
        when {
            params {
                aligner = "bismark_hisat"
                skip_trimming = true
                rrbs = true
                // Generated by previous test
                bismark_index = ".nf-test/indexes/BismarkIndex/"
                local_alignment = true
                outdir = "$outputDir"
            }
        }

        then {
            assert workflow.success
        }
    }

}
