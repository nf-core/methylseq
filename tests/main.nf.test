import com.askimed.nf.test.util.FileUtil;

nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "methylseq"

    test("METHYLSEQ Workflow main.nf") {

        when {
            params {
                save_reference = true
                outdir = "$outputDir"
            }
        }

        then {
            // Assert Workflow Success
            assert workflow.success

            // Bismark Methylation Calls Splitting Report
            assert snapshot(path("$outputDir/bismark/methylation_calls/splitting_report/Ecoli_10K_methylated_1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt"),
                            path("$outputDir/bismark/methylation_calls/splitting_report/SRR389222_sub1_trimmed_bismark_bt2.deduplicated_splitting_report.txt"),
                            path("$outputDir/bismark/methylation_calls/splitting_report/SRR389222_sub2_trimmed_bismark_bt2.deduplicated_splitting_report.txt")).match("splitting_report")

            // Bismark Summary Report
            assert snapshot(path("$outputDir/bismark/summary/bismark_summary_report.txt")).match("summary")

            // Reference Genome
            assert snapshot(path("$outputDir/bismark/reference_genome/BismarkIndex/genome.fa")).match("reference_genome")

            // Qualimap Report
            assert snapshot(path("$outputDir/qualimap/Ecoli_10K_methylated/genome_results.txt"),
                            path("$outputDir/qualimap/SRR389222_sub1/genome_results.txt"),
                            path("$outputDir/qualimap/SRR389222_sub2/genome_results.txt")).match("qualimap")

            // Software Versions
            assert snapshot(path("$outputDir/pipeline_info/software_versions.yml")).match("software_versions")
        }
    }

}

