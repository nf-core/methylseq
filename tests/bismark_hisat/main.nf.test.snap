{
    "single_end": {
        "content": [
            "Ecoli_10K_methylated_1_val_1_bismark_hisat2_pe.deduplicated_splitting_report.txt:md5,4e3ab3792be5eedde05d468a7e7f17e4",
            "SRR389222_sub1_trimmed_bismark_hisat2.deduplicated_splitting_report.txt:md5,4020063dd8610b58b7b39c3ca62cef17",
            "SRR389222_sub2_trimmed_bismark_hisat2.deduplicated_splitting_report.txt:md5,4db2155fdfae1cb9c7c758edba256e2e",
            "genome_results.txt:md5,cd5d3b03821c038eb3bda24beb3ae938",
            "genome_results.txt:md5,9956ed1dcf51e6975f4a6ab8fe156aba",
            "genome_results.txt:md5,1133ea5d1376389c6a17def89d2e90ec",
            "software_versions.yml:md5,f281f6db0fb109fd50575688149859e3"
        ],
        "timestamp": "2023-04-25T16:37:08+0000"
    },
    "single_end_rrbs": {
        "content": [
            "Ecoli_10K_methylated_R1_bismark_hisat2_pe_splitting_report.txt:md5,f59c6bf8089c7dde3225cc577e0260e8",
            "SRR389222_sub1_bismark_hisat2_splitting_report.txt:md5,754ed7bf4d75acee466c48c05a11d331",
            "SRR389222_sub2.merged_bismark_hisat2_splitting_report.txt:md5,0629e1f8b4727efe8fd69500783eec61",
            "genome_results.txt:md5,3afd1eb7d4dc72a05e4d4de887d791e3",
            "genome_results.txt:md5,58a2aeeb5eacca4c766bb23fee8f9013",
            "genome_results.txt:md5,473022a29cf12d97e75451a4eea67c02",
            "software_versions.yml:md5,986af83c1bf68311b22432ed217161bd"
        ],
        "timestamp": "2023-04-25T16:37:08+0000"
    }
}